### ISSUE TITLE
Question: is there no API exposed to manipulate the WinUIDesktopWin32WindowClass window size?

### ISSUE THREAD

**Issue by mqudsi** [microsoft/microsoft-ui-xaml#2564]
For a C#/Win32 project using XAML Islands to host WinUI 3, there does not seem to be any API that I can find for interacting with the parent window? e.g. without manually obtaining the hWnd and using the native api, it seems that you have access to the "fake" CoreWindow but no access to the actual top-level Window.

For example, with ApplicationView.SetPreferredMinSize and ApplicationView.TryResizeView not working, I wanted to work around it by manually setting the window size. While XamlRoot can be used to get the content size, it can't be used to set it.

**Comment by mdtauk** [comment]
There is a Window element, but with Win32 it links to the HWND, and with UWP it links to the CoreWindow.

The future direction and capabilities of Windowing is an ongoing topic, and Win32 enables you to go into the HWND, but UWP doesn't allow this.

**Comment by mqudsi** [comment]
Is there an abstraction over the two to set the window size?

**Comment by mdtauk** [comment]
> Is there an abstraction over the two to set the window size?

This is the spec as it is being developed AFAICT
https://github.com/microsoft/microsoft-ui-xaml-specs/blob/Win32/active/Win32/Window_and_Application_API_Spec.md

**Comment by mqudsi** [comment]
Thanks. I see it has this 

> Win32: The Window is created with the default size of the Window specify by the Window Shell.
> UWP: Keep the current behavior. To change the size of the CoreWindow in UWP, it’s required to use ApplicationView’s PreferredLaunchViewSize and PreferredLaunchWindowingMode.

I guess I should open an issue there? Because without additional APIs to support the Win32 case, there's a disparity in how much control the two different window modes give the application/developer (and ironically, uwp is giving more control than win32 in this case).

Then again, the entire `ApplicationView` class is not under the purview of WinUI (it still lives under `Windows.UI`), so all WinUI would have to do is expose the win32 feature the same way it exposes the UWP feature.

**Comment by ranjeshj** [comment]
@marb2000 as FYI

**Comment by marb2000** [comment]
In Preview 1 there is not an API that allows you to set the window size for Desktop. It's planned for future previews (TBD yet). 

If you want to change the Window's size you can use Win32 APIs. For example using the SetWindowPos:

```cs
[DllImport("user32.dll", SetLastError = true)]
public static extern bool SetWindowPos(IntPtr hWnd, IntPtr hWndInsertAfter, int X, int Y, int cx, int cy, int uFlags);
```
Then you can move it using the flag SWP_SHOWWINDOW

```cs
int HWND_TOP = 0;
int SWP_SHOWWINDOW = 0x0040;
SetWindowPos(windowHandle, (IntPtr)HWND_TOP, position_x, position_y, window_width, window_height,SWP_SHOWWINDOW);
```
To get Window Handle you can you use the [IWindowNative](https://github.com/microsoft/WinUI-3-Demos/blob/0bc934a6716235a7f55e3f2a90f4dbb045a0697d/DemoBuildCs/DemoBuildCs/DemoBuildCs/App.xaml.cs#L46) COM interface that I used in my [//Build demo](https://github.com/microsoft/WinUI-3-Demos):

```cs
Window window = new MainWindow();
WindowWrapper windowWrapper = WindowWrapper.FromAbi(window.ThisPtr);
IntPtr windowHandle = windowWrapper.WindowHandle;
```
Hope this helps meanwhile.

**Comment by mqudsi** [comment]
Thanks, @marb2000 both for the future API info and especially for the `IWindowNative` info. I had already tried to use the native win32 APIs but was unable to figure out how to get the hWnd without digging through the process memory. I thought the abstraction over the two window types was in CLR and not in WinRT/COM, so I was barking up the wrong tree with my approach.

Note that it seems you must first activate the window or else XamlUdk will throw an exception on the call, which is unfortunate because it means that there's a flash of the wrong window size before it gets set correctly.

**Comment by oliverw** [comment]
2023 and still no such feature in sight?

**Comment by pratikone** [comment]
You may want to explore the new `Window.AppWindow` api we have released as part of WinAppSDK 1.3 preview. It simplifies a lot of HWND related operations.
API : https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.windowing.appwindow?view=windows-app-sdk-1.1
https://github.com/microsoft/microsoft-ui-xaml/blob/main/specs/appwindow-spec.md is the spec
